src/WriterRegistrar.cpp

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#include "WriterRegistrar.h"
+
+namespace WriterModule {
+namespace Registry {
+
+struct FactoryInfo {
+  ModuleFactory FactoryPtr;
+  std::string Id;
+  std::string Name;
+};
+
+std::map<IdNameHash, FactoryInfo> &getFactories() {
+  static std::map<IdNameHash, FactoryInfo> Factories;
+  return Factories;
+}
+
+IdNameHash getWriterModuleHash(std::string const &FlatbufferID,
+                               std::string const &ModuleName) {
+  return std::hash<std::string>{}(FlatbufferID + ModuleName);
+}
+
+std::map<std::string, std::string> getFactoryIdsAndNames() {
+  std::map<std::string, std::string> ReturnMap;
+  auto const Factories = getFactories();
+  for (auto const &Item : Factories) {
+    ReturnMap.insert({Item.second.Id, Item.second.Name});
+  }
+  return ReturnMap;
+}
+
+ModuleFactory const &find(std::string const &FlatbufferID) {
+  auto const &Factories = getFactories();
+  auto FoundFactory = std::find_if(Factories.begin(), Factories.end(),
+                                   [&FlatbufferID](auto const &CItem) {
+                                     return CItem.second.Id == FlatbufferID;
+                                   });
+  if (FoundFactory == Factories.end()) {
+    throw std::out_of_range("Unable to find factory with the id: " +
+                            FlatbufferID);
+  }
+  return (*FoundFactory).second.FactoryPtr;
+}
+
+ModuleFactory const &find(std::string const &FlatbufferID,
+                          std::string const &ModuleName) {
+  auto const &Factories = getFactories();
```

Comment:
Check `if (not FlatbufferID.empty())` at the top as a guard condition.
It is empty then we aren't going to get anything useful, right?